File: FoxTales/FoxTales.Domain/Entities/Question.cs
using FoxTales.Domain.Enums;

namespace FoxTales.Domain.Entities;

public class Question
{
    public int? Id { get; set; }
    public required string Text { get; set; }
    public bool IsPublic { get; set; }
    public required Language Language { get; set; }
    public DateTime CreatedDate { get; set; }

    public int OwnerId { get; set; }
    public virtual User Owner { get; set; } = null!;
}

File: src/models/Question.ts
import { GENDER } from "@/enums/userEnum";
import { LANG } from "@/enums/languagesEnum";

export class Question {
  id: number | null;
  text: string;
  ownerId: number;
  language: LANG;
  targetGender: GENDER;
  isPublic: boolean;
  usedCount: number;
  publicDate: Date | null;
  createdDate: Date;
  catalogIds: number[] = [];

  constructor(
    id: number | null = null,
    text: string = "",
    ownerId: number = 0,
    language: LANG = LANG.PL,
    targetGender: GENDER = GENDER.ALL,
    isPublic: boolean = false,
    usedCount: number = 0,
    publicDate: Date | null = null,
    createdDate: Date = new Date()
  ) {
    this.id = id;
    this.text = text;
    this.isPublic = isPublic;
    this.ownerId = ownerId;
    this.language = language;
    this.targetGender = targetGender;
    this.usedCount = usedCount;
    this.publicDate = publicDate;
    this.createdDate = createdDate;
  }

  addCatalogs(catalogIds: number[]): void {
    this.catalogIds = catalogIds;
  }
}
